samples/snippets/test_get_operation.py

```diff
@@ -0,0 +1,80 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import google.auth
+
+from google.cloud import contact_center_insights_v1
+
+import pytest
+
+import get_operation
+
+TRANSCRIPT_URI = "gs://cloud-samples-data/ccai/chat_sample.json"
+AUDIO_URI = "gs://cloud-samples-data/ccai/voice_6912.txt"
+
+
+@pytest.fixture
+def project_id():
+    _, project_id = google.auth.default()
+    return project_id
+
+
+@pytest.fixture
+def insights_client():
+    return contact_center_insights_v1.ContactCenterInsightsClient()
+
+
+@pytest.fixture
+def conversation_resource(project_id, insights_client):
+    # Create a conversation.
+    parent = contact_center_insights_v1.ContactCenterInsightsClient.common_location_path(
+        project_id, "us-central1"
+    )
+
+    conversation = contact_center_insights_v1.Conversation()
+    conversation.data_source.gcs_source.transcript_uri = TRANSCRIPT_URI
+    conversation.data_source.gcs_source.audio_uri = AUDIO_URI
+    conversation.medium = contact_center_insights_v1.Conversation.Medium.CHAT
+
+    conversation = insights_client.create_conversation(
+        parent=parent, conversation=conversation
+    )
+    yield conversation
+
+    # Delete the conversation.
+    delete_request = contact_center_insights_v1.DeleteConversationRequest()
+    delete_request.name = conversation.name
+    delete_request.force = True
+    insights_client.delete_conversation(request=delete_request)
+
+
+@pytest.fixture
+def analysis_operation(conversation_resource, insights_client):
+    # Create an analysis.
+    conversation_name = conversation_resource.name
+    analysis = contact_center_insights_v1.Analysis()
+    analysis_operation = insights_client.create_analysis(
+        parent=conversation_name, analysis=analysis
+    )
+
+    # Wait until the analysis operation is done and return the operation.
+    analysis_operation.result(timeout=86400)
```

Comment:
Can you please reduce the timout here? Looking at one of the tests, it takes the entire test approx 30 seconds to complete (https://screenshot.googleplex.com/6Z5n6rJGtAZcWUN). Maybe 60 seconds is sufficient?

Reply:
I reduced this to 10 minutes to prevent flaky test. Although this test took about 30 seconds, Insights' SLO for creating an analysis is a few hours because we're going to extend this API to do more complex analyses.